app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -288,16 +288,29 @@ class NavigationOverlayFragment : Fragment() {
                         fxaButton.contentDescription =
                             resources.getString(R.string.fxa_navigation_item_new,
                                 resources.getString(R.string.app_name))
+                        resetFxaOnboardingShown()
                     }
                     AccountState.NeedsReauthentication -> {
                         fxaButton.setImageResource(R.drawable.ic_fxa_needs_reauthentication)
                         fxaButton.contentDescription =
                             resources.getString(R.string.fxa_navigation_item_sign_in_again)
+                        resetFxaOnboardingShown()
                     }
                 }
             }
     }
 
+    /**
+     * We need to reset the state of FxA onboarding screen to ensure that we do not show onboarding on every startup (#2861).
+     * It will get reset whenever there are FxA authentication problems or the user logs out.
```

Comment:
This line is redundant to the code (and separated from where it actually happens so this comment is unlkely to stay up to date) so I'd remove it.